File: RootTools2/src/main/java/com/rarnu/tools/neo/fragment/MIUIAppSettingFragment.java
package com.rarnu.tools.neo.fragment;

import android.content.SharedPreferences;
import android.os.Build;
import android.os.Bundle;
import android.preference.Preference;
import android.view.Menu;
import com.rarnu.tools.neo.R;
import com.rarnu.tools.neo.api.DeviceAPI;
import com.rarnu.tools.neo.base.BasePreferenceFragment;
import com.rarnu.tools.neo.comp.PreferenceEx;
import com.rarnu.tools.neo.xposed.XpStatus;

/**
 * Created by rarnu on 11/23/16.
 */
public class MIUIAppSettingFragment extends BasePreferenceFragment implements Preference.OnPreferenceClickListener {

    private PreferenceEx pBrowser, pCalendar, pCleanMaster, pDownload, pFileExplorer, pMms, pMusic, pSearchBox, pVideo, pWeather, pSystem;
    private SharedPreferences pref = null;
    private SharedPreferences.Editor editor = null;

    @Override
    public int getBarTitle() {
        return R.string.miui_app_name;
    }

    @Override
    public String getCustomTitle() {
        return null;
    }

    @Override
    public void initComponents() {
        pref = getContext().getSharedPreferences(XpStatus.PREF, Build.VERSION.SDK_INT < 24 ? 1 : 0);
        editor = pref.edit();
        pBrowser = findPref(R.string.id_app_browser);
        pCalendar = findPref(R.string.id_app_calendar);
        pCleanMaster = findPref(R.string.id_app_cleanmaster);
        pDownload = findPref(R.string.id_app_download);
        pFileExplorer = findPref(R.string.id_app_fileexplorer);
        pMms = findPref(R.string.id_app_mms);
        pMusic = findPref(R.string.id_app_music);
        pSearchBox = findPref(R.string.id_app_searchbox);
        pVideo = findPref(R.string.id_app_video);
        pWeather = findPref(R.string.id_app_weather);
        pSystem = findPref(R.string.id_app_system);
    }

    private PreferenceEx findPref(int prefId) {
        return (PreferenceEx) findPreference(getString(prefId));
    }

    @Override
    public void initEvents() {
        pBrowser.setOnPreferenceClickListener(this);
        pCalendar.setOnPreferenceClickListener(this);
        pCleanMaster.setOnPreferenceClickListener(this);
        pDownload.setOnPreferenceClickListener(this);
        pFileExplorer.setOnPreferenceClickListener(this);
        pMms.setOnPreferenceClickListener(this);
        pMusic.setOnPreferenceClickListener(this);
        pSearchBox.setOnPreferenceClickListener(this);
        pVideo.setOnPreferenceClickListener(this);
        pWeather.setOnPreferenceClickListener(this);
        pSystem.setOnPreferenceClickListener(this);
    }

    @Override
    public void initLogic() {
        pBrowser.setStatus(pref.getBoolean(XpStatus.KEY_AD_BROWSER, false));
        pCalendar.setStatus(pref.getBoolean(XpStatus.KEY_AD_CALENDAR, false));
        pCleanMaster.setStatus(pref.getBoolean(XpStatus.KEY_AD_CLEANMASTER, false));
        pDownload.setStatus(pref.getBoolean(XpStatus.KEY_AD_DOWNLOAD, false));
        pFileExplorer.setStatus(pref.getBoolean(XpStatus.KEY_AD_FILEEXPLORER, false));
        pMms.setStatus(pref.getBoolean(XpStatus.KEY_AD_MMS, false));
        pMusic.setStatus(pref.getBoolean(XpStatus.KEY_AD_MUSIC, false));
        pSearchBox.setStatus(pref.getBoolean(XpStatus.KEY_AD_SEARCHBOX, false));
        pVideo.setStatus(pref.getBoolean(XpStatus.KEY_AD_VIDEO, false));
        pWeather.setStatus(pref.getBoolean(XpStatus.KEY_AD_WEATHER, false));
        pSystem.setStatus(pref.getBoolean(XpStatus.KEY_AD_SYSTEM, false));
    }

    @Override
    public int getFragmentLayoutResId() {
        return R.xml.miui_apps;
    }

    @Override
    public String getMainActivityName() {
        return null;
    }

    @Override
    public void initMenu(Menu menu) {

    }

    @Override
    public void onGetNewArguments(Bundle bn) {

    }

    @Override
    public Bundle getFragmentState() {
        return null;
    }

    @Override
    public boolean onPreferenceClick(Preference preference) {
        String prefKey = preference.getKey();
        PreferenceEx ex = (PreferenceEx) preference;
        if (prefKey.equals(getString(R.string.id_app_browser))) {
            ex.setStatus(!ex.getStatus());
            editor.putBoolean(XpStatus.KEY_AD_BROWSER, ex.getStatus()).apply();
            DeviceAPI.makePreferenceReadable(Build.VERSION.SDK_INT, getContext().getPackageName());
        } else if (prefKey.equals(getString(R.string.id_app_calendar))) {
            ex.setStatus(!ex.getStatus());
            editor.putBoolean(XpStatus.KEY_AD_CALENDAR, ex.getStatus()).apply();
            DeviceAPI.makePreferenceReadable(Build.VERSION.SDK_INT, getContext().getPackageName());
        } else if (prefKey.equals(getString(R.string.id_app_cleanmaster))) {
            ex.setStatus(!ex.getStatus());
            editor.putBoolean(XpStatus.KEY_AD_CLEANMASTER, ex.getStatus()).apply();
            DeviceAPI.makePreferenceReadable(Build.VERSION.SDK_INT, getContext().getPackageName());
        } else if (prefKey.equals(getString(R.string.id_app_download))) {
            ex.setStatus(!ex.getStatus());
            editor.putBoolean(XpStatus.KEY_AD_DOWNLOAD, ex.getStatus()).apply();
            DeviceAPI.makePreferenceReadable(Build.VERSION.SDK_INT, getContext().getPackageName());
        } else if (prefKey.equals(getString(R.string.id_app_fileexplorer))) {
            ex.setStatus(!ex.getStatus());
            editor.putBoolean(XpStatus.KEY_AD_FILEEXPLORER, ex.getStatus()).apply();
            DeviceAPI.makePreferenceReadable(Build.VERSION.SDK_INT, getContext().getPackageName());
        } else if (prefKey.equals(getString(R.string.id_app_mms))) {
            ex.setStatus(!ex.getStatus());
            editor.putBoolean(XpStatus.KEY_AD_MMS, ex.getStatus()).apply();
            DeviceAPI.makePreferenceReadable(Build.VERSION.SDK_INT, getContext().getPackageName());
        } else if (prefKey.equals(getString(R.string.id_app_music))) {
            ex.setStatus(!ex.getStatus());
            editor.putBoolean(XpStatus.KEY_AD_MUSIC, ex.getStatus()).apply();
            DeviceAPI.makePreferenceReadable(Build.VERSION.SDK_INT, getContext().getPackageName());
        } else if (prefKey.equals(getString(R.string.id_app_searchbox))) {
            ex.setStatus(!ex.getStatus());
            editor.putBoolean(XpStatus.KEY_AD_SEARCHBOX, ex.getStatus()).apply();
            DeviceAPI.makePreferenceReadable(Build.VERSION.SDK_INT, getContext().getPackageName());
        } else if (prefKey.equals(getString(R.string.id_app_video))) {
            ex.setStatus(!ex.getStatus());
            editor.putBoolean(XpStatus.KEY_AD_VIDEO, ex.getStatus()).apply();
            DeviceAPI.makePreferenceReadable(Build.VERSION.SDK_INT, getContext().getPackageName());
        } else if (prefKey.equals(getString(R.string.id_app_weather))) {
            ex.setStatus(!ex.getStatus());
            editor.putBoolean(XpStatus.KEY_AD_WEATHER, ex.getStatus()).apply();
            DeviceAPI.makePreferenceReadable(Build.VERSION.SDK_INT, getContext().getPackageName());
        } else if (prefKey.equals(getString(R.string.id_app_system))) {
            ex.setStatus(!ex.getStatus());
            editor.putBoolean(XpStatus.KEY_AD_SYSTEM, ex.getStatus()).apply();
            DeviceAPI.makePreferenceReadable(Build.VERSION.SDK_INT, getContext().getPackageName());
        }
        return true;
    }
}

File: RootTools2/src/main/kotlin/com/rarnu/tools/neo/fragment/MIUIAppSettingFragment.kt
package com.rarnu.tools.neo.fragment

import android.content.SharedPreferences
import android.os.Build
import android.os.Bundle
import android.preference.Preference
import android.view.Menu
import com.rarnu.tools.neo.R
import com.rarnu.tools.neo.api.DeviceAPI
import com.rarnu.tools.neo.base.BasePreferenceFragment
import com.rarnu.tools.neo.comp.PreferenceEx
import com.rarnu.tools.neo.xposed.XpStatus

/**
 * Created by rarnu on 11/23/16.
 */
class MIUIAppSettingFragment : BasePreferenceFragment(), Preference.OnPreferenceClickListener {

    private var pAccount: PreferenceEx? = null
    private var pBrowser: PreferenceEx? = null
    private var pCalendar: PreferenceEx? = null
    private var pCleanMaster: PreferenceEx? = null
    private var pDownload: PreferenceEx? = null
    private var pFileExplorer: PreferenceEx? = null
    private var pContact: PreferenceEx? = null
    private var pMms: PreferenceEx? = null
    private var pMusic: PreferenceEx? = null
    private var pSearchBox: PreferenceEx? = null
    private var pVideo: PreferenceEx? = null
    private var pWeather: PreferenceEx? = null
    private var pTheme: PreferenceEx? = null
    private var pMarket: PreferenceEx? = null
    private var pCloudService: PreferenceEx? = null
    private var pSettings: PreferenceEx? = null
    private var pSystem: PreferenceEx? = null
    private var pref: SharedPreferences? = null
    private var editor: SharedPreferences.Editor? = null

    override fun getBarTitle(): Int = R.string.miui_app_name

    override fun getCustomTitle(): String? = null

    override fun initComponents() {
        pref = context?.getSharedPreferences(XpStatus.PREF, if (Build.VERSION.SDK_INT < 24) 1 else 0)
        editor = pref?.edit()
        pAccount = findPref(R.string.id_app_account)
        pBrowser = findPref(R.string.id_app_browser)
        pCalendar = findPref(R.string.id_app_calendar)
        pCleanMaster = findPref(R.string.id_app_cleanmaster)
        pDownload = findPref(R.string.id_app_download)
        pFileExplorer = findPref(R.string.id_app_fileexplorer)
        pContact = findPref(R.string.id_app_contacts)
        pMms = findPref(R.string.id_app_mms)
        pMusic = findPref(R.string.id_app_music)
        pSearchBox = findPref(R.string.id_app_searchbox)
        pVideo = findPref(R.string.id_app_video)
        pWeather = findPref(R.string.id_app_weather)
        pTheme = findPref(R.string.id_app_thememanager)
        pMarket = findPref(R.string.id_app_market)
        pCloudService = findPref(R.string.id_app_cloudservice)
        pSettings = findPref(R.string.id_app_settings)
        pSystem = findPref(R.string.id_app_system)
    }

    private fun findPref(prefId: Int): PreferenceEx = findPreference(getString(prefId)) as PreferenceEx

    override fun initEvents() {
        pAccount?.onPreferenceClickListener = this
        pBrowser?.onPreferenceClickListener = this
        pCalendar?.onPreferenceClickListener = this
        pCleanMaster?.onPreferenceClickListener = this
        pDownload?.onPreferenceClickListener = this
        pFileExplorer?.onPreferenceClickListener = this
        pContact?.onPreferenceClickListener = this
        pMms?.onPreferenceClickListener = this
        pMusic?.onPreferenceClickListener = this
        pSearchBox?.onPreferenceClickListener = this
        pVideo?.onPreferenceClickListener = this
        pWeather?.onPreferenceClickListener = this
        pTheme?.onPreferenceClickListener = this
        pMarket?.onPreferenceClickListener = this
        pCloudService?.onPreferenceClickListener = this
        pSettings?.onPreferenceClickListener = this
        pSystem?.onPreferenceClickListener = this
    }

    override fun initLogic() {
        pAccount?.status = pref!!.getBoolean(XpStatus.KEY_AD_ACCOUNT, false)
        pBrowser?.status = pref!!.getBoolean(XpStatus.KEY_AD_BROWSER, false)
        pCalendar?.status = pref!!.getBoolean(XpStatus.KEY_AD_CALENDAR, false)
        pCleanMaster?.status = pref!!.getBoolean(XpStatus.KEY_AD_CLEANMASTER, false)
        pDownload?.status = pref!!.getBoolean(XpStatus.KEY_AD_DOWNLOAD, false)
        pFileExplorer?.status = pref!!.getBoolean(XpStatus.KEY_AD_FILEEXPLORER, false)
        pContact?.status = pref!!.getBoolean(XpStatus.KEY_AD_CONTACTS, false)
        pMms?.status = pref!!.getBoolean(XpStatus.KEY_AD_MMS, false)
        pMusic?.status = pref!!.getBoolean(XpStatus.KEY_AD_MUSIC, false)
        pSearchBox?.status = pref!!.getBoolean(XpStatus.KEY_AD_SEARCHBOX, false)
        pVideo?.status = pref!!.getBoolean(XpStatus.KEY_AD_VIDEO, false)
        pWeather?.status = pref!!.getBoolean(XpStatus.KEY_AD_WEATHER, false)
        pTheme?.status = pref!!.getBoolean(XpStatus.KEY_AD_THEMEMANAGER, false)
        pMarket?.status = pref!!.getBoolean(XpStatus.KEY_AD_MARKET, false)
        pCloudService?.status = pref!!.getBoolean(XpStatus.KEY_AD_CLOUDSERVICE, false)
        pSettings?.status = pref!!.getBoolean(XpStatus.KEY_AD_SETTINGS, false)
        pSystem?.status = pref!!.getBoolean(XpStatus.KEY_AD_SYSTEM, false)
    }

    override fun getFragmentLayoutResId(): Int = R.xml.miui_apps

    override fun getMainActivityName(): String? = null

    override fun initMenu(menu: Menu?) { }

    override fun onGetNewArguments(bn: Bundle?) { }

    override fun getFragmentState(): Bundle? = null

    override fun onPreferenceClick(preference: Preference): Boolean {
        val prefKey = preference.key
        val ex = preference as PreferenceEx
        ex.status = !ex.status
        when (prefKey) {
            getString(R.string.id_app_account) -> editor?.putBoolean(XpStatus.KEY_AD_ACCOUNT, ex.status)
            getString(R.string.id_app_browser) -> editor?.putBoolean(XpStatus.KEY_AD_BROWSER, ex.status)
            getString(R.string.id_app_calendar) -> editor?.putBoolean(XpStatus.KEY_AD_CALENDAR, ex.status)
            getString(R.string.id_app_cleanmaster) -> editor?.putBoolean(XpStatus.KEY_AD_CLEANMASTER, ex.status)
            getString(R.string.id_app_download) -> editor?.putBoolean(XpStatus.KEY_AD_DOWNLOAD, ex.status)
            getString(R.string.id_app_fileexplorer) -> editor?.putBoolean(XpStatus.KEY_AD_FILEEXPLORER, ex.status)
            getString(R.string.id_app_contacts) -> editor?.putBoolean(XpStatus.KEY_AD_CONTACTS, ex.status)
            getString(R.string.id_app_mms) -> editor?.putBoolean(XpStatus.KEY_AD_MMS, ex.status)
            getString(R.string.id_app_music) -> editor?.putBoolean(XpStatus.KEY_AD_MUSIC, ex.status)
            getString(R.string.id_app_searchbox) -> editor?.putBoolean(XpStatus.KEY_AD_SEARCHBOX, ex.status)
            getString(R.string.id_app_video) -> editor?.putBoolean(XpStatus.KEY_AD_VIDEO, ex.status)
            getString(R.string.id_app_weather) -> editor?.putBoolean(XpStatus.KEY_AD_WEATHER, ex.status)
            getString(R.string.id_app_thememanager) -> editor?.putBoolean(XpStatus.KEY_AD_THEMEMANAGER, ex.status)
            getString(R.string.id_app_market) -> editor?.putBoolean(XpStatus.KEY_AD_MARKET, ex.status)
            getString(R.string.id_app_cloudservice) -> editor?.putBoolean(XpStatus.KEY_AD_CLOUDSERVICE, ex.status)
            getString(R.string.id_app_settings) -> editor?.putBoolean(XpStatus.KEY_AD_SETTINGS, ex.status)
            getString(R.string.id_app_system) -> editor?.putBoolean(XpStatus.KEY_AD_SYSTEM, ex.status)
        }
        editor?.apply()
        DeviceAPI.makePreferenceReadable(Build.VERSION.SDK_INT, context?.packageName)
        return true
    }
}
